### DISCUSSION TITLE
How to deploy NodeJS function with code in several files?

### DISCUSSION THREAD

**Discussion by sahilsachdeva** [fission/fission#2149] [Q&A] [answered]
The NodeJS documentation gives a good example to deploy a function with publicly available package.

But how can a function with code spread over multiple files be deployed?

I am trying to deploy the following:

package.json - with publicly available package listed as dependency
function.js - uses the publicly available dependency 
index.js - depends on function.js

I always get `Error: error getting function response`

Additionally I think the way to package and deploy mention here #2031 is more elegant with zip files

**Comment by sanketsudake** [comment, marked as answer]
Hey @sahilsachdeva,

Fission provides builder and environment. You can create a source zip and give it Fission function. Building of the function can be done with Fission builder.
Fission provide package abstraction which is used combine multiple files.

References:
1. https://github.com/fission/examples/tree/master/nodejs
2. Please check package reference here https://docs.fission.io/docs/usage/languages/nodejs/
